infra/gcp/terraform/k8s-infra-oci-proxy/network.tf

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "google_compute_global_address" "default_ipv4" {
+  project      = google_project.project.project_id
+  name         = "k8s-infra-oci-proxy-sandbox"
+  address_type = "EXTERNAL"
+  ip_version   = "IPV4"
+}
+
+resource "google_compute_global_address" "default_ipv6" {
+  project      = google_project.project.project_id
+  name         = "k8s-infra-oci-proxy-sandbox-v6"
+  address_type = "EXTERNAL"
+  ip_version   = "IPV6"
+}
+
+
+resource "google_compute_region_network_endpoint_group" "default" {
+  for_each = google_cloud_run_service.regions
+
+  provider              = google-beta
+  project               = google_project.project.project_id
+  name                  = "${local.project_id}--${each.key}--neg"
+  network_endpoint_type = "SERVERLESS"
+  region                = google_cloud_run_service.regions[each.key].location
+  cloud_run {
+    service = google_cloud_run_service.regions[each.key].name
+  }
+}
+
+module "lb-http" {
+  source  = "GoogleCloudPlatform/lb-http/google//modules/serverless_negs"
+  version = "~> 6.2.0"
+
+  project = google_project.project.project_id
+  name    = local.project_id
+
+  # ...
+  backends = {
+    default = {
+      description = null
+      groups = [
+        for neg in google_compute_region_network_endpoint_group.default :
+        {
+          group = neg.id
+        }
+      ]
+      enable_cdn              = true
+      security_policy         = null
+      custom_request_headers  = null
+      custom_response_headers = null
+
+      iap_config = {
+        enable               = false
+        oauth2_client_id     = ""
+        oauth2_client_secret = ""
+      }
+
+      log_config = {
+        enable      = true
+        sample_rate = "1.0"
+      }
+    }
+  }
+  create_address      = false
+  create_ipv6_address = false
+  enable_ipv6         = true
+  https_redirect      = true
+  #TODO(ameukam): current the TF resource google_compute_global_address don't have
+  #the value of IP in his attribute. But it's accessible with the data source:
+  #https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/compute_global_address
+  address      = "34.110.128.221"
```

Comment:
this will have an unfortunate bootstrapping problem on a future clean bring-up, but that's probably ok for now. 

To avoid it, although it _kinda_ sucks from a guaranteeing convergence perspective in many cases, we could have a data source that depends on the IP address creation to look these up (Or we end up refactoring this to work in multiple steps).